dangerzone/gui/main_window.py

```diff
@@ -459,38 +482,75 @@ def check_state(self) -> None:
             state = "not_installed"
 
         else:
-            # Can we run `docker image ls` without an error
+            # Can we run `docker/podman image ls` without an error
             with subprocess.Popen(
                 [container_runtime, "image", "ls"],
                 stdout=subprocess.DEVNULL,
-                stderr=subprocess.DEVNULL,
+                stderr=subprocess.PIPE,
                 startupinfo=get_subprocess_startupinfo(),
             ) as p:
-                p.communicate()
+                _, stderr = p.communicate()
                 if p.returncode != 0:
-                    log.error("Docker is not running")
+                    runtime_name = self.dangerzone.isolation_provider.get_runtime_name()
+                    log.error(f"{runtime_name} is not running")
                     state = "not_running"
+                    error = stderr
                 else:
                     # Always try installing the container
                     state = "install_container"
 
         # Update the state
-        self.state_change(state)
+        self.state_change(state, error)
 
-    def state_change(self, state: str) -> None:
+    def state_change(self, state: str, error: str) -> None:
         if state == "not_installed":
-            self.label.setText(
-                "<strong>Dangerzone Requires Docker Desktop</strong><br><br><a href='https://www.docker.com/products/docker-desktop'>Download Docker Desktop</a>, install it, and open it."
-            )
+            if platform.system() == "Linux":
+                self.label.setText(
+                    (
+                        "<strong>Dangerzone requires Podman</strong><br><br>"
+                        "Install it and retry."
+                    )
+                )
+            else:
+                self.label.setText(
+                    (
+                        "<strong>Dangerzone requires Docker Desktop</strong><br><br>"
+                        "<a href='https://www.docker.com/products/docker-desktop'>Download Docker Desktop</a>"
+                        ", install it, and open it."
+                    )
+                )
             self.buttons.show()
         elif state == "not_running":
-            self.label.setText(
-                "<strong>Dangerzone Requires Docker Desktop</strong><br><br>Docker is installed but isn't running.<br><br>Open Docker and make sure it's running in the background."
-            )
+            if platform.system() == "Linux":
+                message = (
+                    "<strong>Dangerzone requires Podman</strong><br><br>"
+                    "Podman is installed but isn't running.<br>"
```

Comment:
Perhaps this is nitpicking, but there is no such thing as "Podman is running" in the same sense as "Docker is running". Docker works with a daemon/client model, where `dockerd` runs persistently in the background and `docker` is a client to it. Podman by contrast has no background daemon. Every invocation of the `podman` command-line does everything it needs to do on the spot and then disappears with no persistent presence on the system.

This "`not_running`" status is derived from the failure of running `podman image list` above, which in this case would indicate that something is wrong about Podman's setup, but not that it is not running.

So I suggest changing this message to something like "Dangerzone requires Podman. Podman is installed but cannot run properly. See errors below."

Reply:
Hey @EtiennePerot, thanks for the comments.

You're correct and thanks for the explanation. That explains why trying to stop "a podman service" doesn't make the following podman commands fail.

I've updated the error message accordingly.